pkg/reconciliation/reconcile_fql.go

```diff
@@ -1,98 +1,53 @@
 package reconciliation
 
 import (
-	"encoding/json"
 	"errors"
-	"fmt"
-	"strconv"
-	"strings"
 
 	"github.com/k8ssandra/cass-operator/pkg/httphelper"
 	"github.com/k8ssandra/cass-operator/pkg/internal/result"
 )
 
-// parseFQLFromConfig parses the DC config field to determine whether FQL should be enabled based on the presence of full_query_logging_options within the cassandra-yaml field.
-// To ease integration into the reconciliation process, it returns (shouldFQLBeEnabled, serverMajorVersion, ReconcileResult) where the ReconcileResult may be result.Error() or result.Continue().
-func parseFQLFromConfig(rc *ReconciliationContext) (bool, int64, result.ReconcileResult) {
+// CheckFullQueryLogging sets FQL enabled or disabled based on the `enableFQL` parameter, and takes serverMajorVersion for additional validation.
+// It calls the NodeMgmtClient which calls the Cassandra management API and returns a result.ReconcileResult.
+func (rc *ReconciliationContext) CheckFullQueryLogging() result.ReconcileResult {
 	dc := rc.GetDatacenter()
-	serverMajorVersion, err := strconv.ParseInt(strings.Split(dc.Spec.ServerVersion, ".")[0], 10, 8)
+	enableFQL, err := dc.FullQueryEnabled()
 	if err != nil {
-		rc.ReqLogger.Error(err, "error parsing server major version. Can't enable full query logging without knowing this")
-		return false, serverMajorVersion, result.Error(err)
+		return result.Error(err)
 	}
-	shouldFQLBeEnabled := false
-	if dc.Spec.Config != nil {
-		var dcConfig map[string]interface{}
-		if err := json.Unmarshal(dc.Spec.Config, &dcConfig); err != nil {
-			rc.ReqLogger.Error(err, "error unmarshalling DC config JSON")
-			return false, serverMajorVersion, result.Error(err)
-		}
-		casYaml, found := dcConfig["cassandra-yaml"]
-		if !found {
-			return false, serverMajorVersion, result.Continue()
-		}
-		casYamlMap, ok := casYaml.(map[string]interface{})
-		if !ok {
-			err := fmt.Errorf("type casting error")
-			rc.ReqLogger.Error(err, "couldn't cast cassandra-yaml value from config to map[string]interface{}")
-			return false, serverMajorVersion, result.Error(err)
+
+	podList, err := rc.listPods(rc.Datacenter.GetClusterLabels())
+	if err != nil {
+		rc.ReqLogger.Error(err, "error listing all pods in the cluster to progress full query logging reconciliation")
+		return result.RequeueSoon(2)
+	}
+	for _, podPtr := range PodPtrsFromPodList(podList) {
+		features, err := rc.NodeMgmtClient.FeatureSet(podPtr)
+		if err != nil {
+			rc.ReqLogger.Error(err, "failed to verify featureset for FQL support")
```

Comment:
Could this error message be "failed to call NodeMgmtClient.FeatureSet()", and could we add the pod IPs or hostnames?

I'm a bit fanatical about informative error messages sorry😅  In this case we aren't failing to verify the featureSet, we just aren't able to get it at all from the HTTP client.

Reply:
The connection error text comes in the err part of the logging. So that's not removed.